pkg/qemu/qemu.go

```diff
@@ -128,6 +131,76 @@ func EnsureDisk(cfg Config) error {
 	return nil
 }
 
+func newQmpClient(cfg Config) (*qmp.SocketMonitor, error) {
+	qmpSock := filepath.Join(cfg.InstanceDir, filenames.QMPSock)
+	qmpClient, err := qmp.NewSocketMonitor("unix", qmpSock, 5*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return qmpClient, nil
+}
+
+func sendHmpCommand(cfg Config, cmd string, tag string) error {
+	qmpClient, err := newQmpClient(cfg)
+	if err != nil {
+		return err
+	}
+	if err := qmpClient.Connect(); err != nil {
+		return err
+	}
+	defer func() { _ = qmpClient.Disconnect() }()
+	rawClient := raw.NewMonitor(qmpClient)
+	logrus.Infof("Sending HMP %s command", cmd)
+	hmc := fmt.Sprintf("%s %s", cmd, tag)
+	_, err = rawClient.HumanMonitorCommand(hmc, nil)
+	return err
+}
+
+func sendImgCommand(cfg Config, args ...string) (string, error) {
+	diffDisk := filepath.Join(cfg.InstanceDir, filenames.DiffDisk)
+	args = append(args, diffDisk)
+	logrus.Debugf("Running qemu-img %v command", args)
+	cmd := exec.Command("qemu-img", args...)
+	b, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return string(b), err
+}
+
+func Del(cfg Config, run bool, tag string) error {
+	if run {
+		return sendHmpCommand(cfg, "delvm", tag)
+	}
+	// -d  deletes a snapshot
+	_, err := sendImgCommand(cfg, "snapshot", "-d", tag)
+	return err
+}
+
+func Save(cfg Config, run bool, tag string) error {
+	if run {
+		return sendHmpCommand(cfg, "savevm", tag)
+	}
+	// -c  creates a snapshot
+	_, err := sendImgCommand(cfg, "snapshot", "-c", tag)
+	return err
+}
+
+func Load(cfg Config, run bool, tag string) error {
+	if run {
+		return sendHmpCommand(cfg, "loadvm", tag)
+	}
+	// -a  applies a snapshot
+	_, err := sendImgCommand(cfg, "snapshot", "-a", tag)
+	return err
+}
+
+func List(cfg Config) (string, error) {
```

Comment:
Perhaps add a comment to clarify that the result is not machine-parsable

Reply:
Good idea, I think the original API just promises a list of strings (lines).

Reply:
I removed the initial heading, so now it at least _looks_ similar to `limactl ls`

```console
$ limactl snapshot create --tag snap alpine
$ limactl snapshot list alpine
ID        TAG                     VM SIZE                DATE       VM CLOCK
1         snap                        0 B 2022-10-16 11:37:23   00:00:00.000
```